shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Address.hs

```diff
@@ -300,7 +300,7 @@ getAddrStakeReference = do
   header <- B.getWord8
   if testBit header byron
     then pure Nothing
-    else skipHash ([] @(HASH crypto)) >> Just <$> getStakeReference header
+    else skipHash ([] @(ADDRHASH crypto)) >> Just <$> getStakeReference header
```

Comment:
There seems to be duplicated functionality in `Shelley.Spec.Ledger.Address.hs` and `Shelley.Spec.Ledger.DeserializeShort.hs`. In particular, there is a method `getAddrStakeReference` in both, and the error that was fixed in #1782 was still present here. Fortunately, the `getAddrStakeReference` method here in `Address` is _only_ used in testing.